### PR TITLE
Update doc comments

### PR DESCRIPTION
**Pull request by wks**
We removed the feature for letting the coordinator call both `stop_all_mutators` and `resume_mutators` since https://github.com/mmtk/mmtk-core/pull/794 and https://github.com/mmtk/mmtk-core/pull/814.  The documentation should reflect that.

### CODE REVIEW

File: src/vm/collection.rs [hunk 1]
```diff
@@ -12,23 +12,25 @@ pub enum GCThreadContext<VM: VMBinding> {
...
-    /// * `tls`: The thread pointer for the GC controller/coordinator.
+    /// * `tls`: The thread pointer for the GC worker.
     /// * `mutator_visitor`: A callback.  Call it with a mutator as argument to notify MMTk that the mutator is ready to be scanned.
     fn stop_all_mutators<F>(tls: VMWorkerThread, mutator_visitor: F)
     where
         F: FnMut(&'static mut Mutator<VM>);
 
     /// Resume all the mutator threads, the opposite of the above. When a GC is finished, MMTk calls this method.
     ///
+    /// This method may be called by any GC worker thread, and may not be the same GC worker thread that called
```
Comment (k-sareen) [inline src/vm/collection.rs:29]: That is not correct. At least as of now. Currently the GC controller thread calls `resume_mutators` in [`EndOfGC`](https://github.com/mmtk/mmtk-core/blob/abd38ec31064e7d52ae2b492a75b4bcc7f7cc554/src/scheduler/gc_work.rs#L262C50-L262C50)

Comment (k-sareen) [reply src/vm/collection.rs:29]: See here as well: https://github.com/mmtk/mmtk-core/blob/abd38ec31064e7d52ae2b492a75b4bcc7f7cc554/src/scheduler/controller.rs#L140-L143

Comment (wks) [reply src/vm/collection.rs:29]: OK. That's true, although it is not intended.  I'll adjust the comment and not mention "worker".

Comment (k-sareen) [reply src/vm/collection.rs:29]: Random aside, I feel like we should rename `stop_all_mutators` to `suspend_mutators` to be in line with `resume_mutators`.

**Review by k-sareen — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**